Mathlib/Probability/Moments/CovarianceBilin.lean

```diff
@@ -0,0 +1,226 @@
+      rwa [ENNReal.essSup_const_mul, ENNReal.toReal_mul, toReal_enorm] at this
+    gcongr
+    · rw [ENNReal.essSup_const_mul]
+      exact ENNReal.mul_ne_top (by simp) h_Lp.eLpNorm_ne_top
+    · exact essSup_mono_ae <| ae_of_all _ L.le_opNorm_enorm
+  have h0 : 0 < p.toReal := by simp [ENNReal.toReal_pos_iff, pos_iff_ne_zero, hp, Ne.lt_top hp_top]
+  suffices ‖L.toLpₗ μ p h_Lp‖
+      ≤ (‖L‖ₑ ^ p.toReal * ∫⁻ x, ‖x‖ₑ ^ p.toReal ∂μ).toReal ^ p.toReal⁻¹ by
+    refine this.trans_eq ?_
+    simp only [ENNReal.toReal_mul]
+    rw [← ENNReal.toReal_rpow, Real.mul_rpow (by positivity) (by positivity),
+      ← Real.rpow_mul (by positivity), mul_inv_cancel₀ h0.ne', Real.rpow_one, toReal_enorm]
+    rw [eLpNorm_eq_lintegral_rpow_enorm (by simp [hp]) hp_top, ENNReal.toReal_rpow]
+    simp
+  rw [Dual.toLpₗ_apply, Lp.norm_toLp, eLpNorm_eq_lintegral_rpow_enorm (by simp [hp]) hp_top]
+  simp only [ENNReal.toReal_ofNat, ENNReal.rpow_ofNat, one_div]
+  refine ENNReal.toReal_le_of_le_ofReal (by positivity) ?_
+  suffices ∫⁻ x, ‖L x‖ₑ ^ p.toReal ∂μ ≤ ‖L‖ₑ ^ p.toReal * ∫⁻ x, ‖x‖ₑ ^ p.toReal ∂μ by
+    rw [← ENNReal.ofReal_rpow_of_nonneg (by positivity) (by positivity)]
+    gcongr
+    rwa [ENNReal.ofReal_toReal]
+    refine ENNReal.mul_ne_top (by simp) ?_
+    have h := h_Lp.eLpNorm_ne_top
+    rw [eLpNorm_eq_lintegral_rpow_enorm (by simp [hp]) hp_top] at h
+    simpa [h0] using h
+  calc ∫⁻ x, ‖L x‖ₑ ^ p.toReal ∂μ
+  _ ≤ ∫⁻ x, ‖L‖ₑ ^ p.toReal * ‖x‖ₑ ^ p.toReal ∂μ := by
+    refine lintegral_mono fun x ↦ ?_
+    rw [← ENNReal.mul_rpow_of_nonneg]
+    swap; · positivity
+    gcongr
+    exact L.le_opNorm_enorm x
+  _ = ‖L‖ₑ ^ p.toReal * ∫⁻ x, ‖x‖ₑ ^ p.toReal ∂μ := by rw [lintegral_const_mul]; fun_prop
+
+end LinearMap
+
+section ContinuousLinearMap
+
+variable {𝕜 : Type*} [RCLike 𝕜] [NormedSpace 𝕜 E] [OpensMeasurableSpace E]
+
+/-- `MemLp.toLp` as a continuous linear map from `Dual 𝕜 E` to `Lp 𝕜 p μ`. -/
+noncomputable
+def toLp (μ : Measure E) (p : ℝ≥0∞) [Fact (1 ≤ p)] (h_Lp : MemLp id p μ) :
+    Dual 𝕜 E →L[𝕜] Lp 𝕜 p μ where
+  toLinearMap := Dual.toLpₗ μ p h_Lp
+  cont := by
+    refine LinearMap.continuous_of_locally_bounded _ fun s hs ↦ ?_
+    rw [image_isVonNBounded_iff]
+    simp_rw [isVonNBounded_iff'] at hs
+    obtain ⟨r, hxr⟩ := hs
+    refine ⟨r * (eLpNorm id p μ).toReal, fun L hLs ↦ ?_⟩
+    specialize hxr L hLs
+    refine (Dual.norm_toLpₗ_le h_Lp L).trans ?_
+    gcongr
+
+@[simp]
+lemma toLp_apply [Fact (1 ≤ p)] (h_Lp : MemLp id p μ) (L : Dual 𝕜 E) :
+    L.toLp μ p h_Lp = MemLp.toLp L (h_Lp.continuousLinearMap_comp L) := rfl
+
+end ContinuousLinearMap
+
+end NormedSpace.Dual
+
+namespace ProbabilityTheory
+
+section Centered
+
+variable [NormedSpace ℝ E] [OpensMeasurableSpace E]
+
+/-- Continuous bilinear form with value `∫ x, L₁ x * L₂ x ∂μ` on `(L₁, L₂)`.
+This is equal to the covariance only if `μ` is centered. -/
+noncomputable
+def centeredCovarianceBilin (μ : Measure E) (h : MemLp id 2 μ) :
```

Comment:
I find this name a little bit strange, because this a covariance without centering term. `noncenteredCovarianceBilin` would look more natural to me. What do you think?